### DISCUSSION TITLE
Apply custom randomization on __getitem__ during training.

### DISCUSSION THREAD

**Discussion by ysig** [huggingface/datasets#5139] [Q&A] [answered]
Hi,

I have an extremely simple use case which I cannot understand how to do with your library.

With `load_dataset` I load a text dataset and I want to apply randomization to the input everytime `__getitem__` (or the equivalent - however it is called) is called on the huggingface dataset object.

(example: __getitem__ normally returns a text and let's say I want to shuffle the words)

I really cannot understand how to do it and I've been searching for more than 2 hours.

Please help!

Thanks,

**Comment by mariosasko** [comment, marked as answer]
Hi! You can use [`set_transform`](https://huggingface.co/docs/datasets/main/en/process#format-transform) for that:
```python

def transform(batch):
    batch["text"] = [shuffle_words(text) for text in batch["text"]]
    return batch

dset.set_transform(transform)
```

**Reply by ysig** [reply]
Thank you so much ::)))
